lib/transport/transport-tls.c

```diff
@@ -135,8 +135,11 @@ log_transport_tls_read_method(LogTransport *s, gpointer buf, gsize buflen, LogTr
               rc = (log_transport_tls_send_shutdown(self) >= 0) ? 0 : -1;
               break;
             case SSL_ERROR_SYSCALL:
-              rc = -1;
-              /* errno is set accordingly */
+              // https://github.com/openssl/openssl/pull/10907
+              // There is a known bug in OpenSSL where it reports SSL_ERROR_SYSCALL without setting
+              // the proper errno value. The mentioned PR were reverted because lot of legacy code
+              // were broken by the fix. OpenSSL 3.0.0 will contain it.
+              rc = (errno == 0) ? 0 : -1;
```

Comment:
I think we should fix the same in `log_transport_tls_write_method()` as well.

Reply:
According to my knowledge write() could not return 0. (Thus not trigger the openssl problem.) However I really like the simmetry behind it, and I am still convincible to implement this change.

Reply:
I think `SSL_ERROR_SYSCALL` with `errno` value of 0 is possible when calling `SSL_write()` as well.

You are right, the interface of `log_transport_write()` is based on the `write()` syscall, which does not return 0 on EOF.
In the above scenario, we could return -1 and set errno ECONNRESET or EPIPE instead.

What do you think?